source/slang/glsl.meta.slang

```diff
@@ -7263,3 +7263,700 @@ __glsl_version(430) [require(glsl)]
+[require(any_gfx_target, shadermemorycontrol_compute)]
+[ForceInline]
+public void memoryBarrierAtomicCounter()
+{
+    __target_switch
+    {
+    case hlsl: __intrinsic_asm "AllMemoryBarrier()";
+    case glsl: __intrinsic_asm "memoryBarrierBuffer()";
+    case spirv:
+    {
+        spirv_asm
+        {
+            OpCapability Shader;
+            OpMemoryBarrier Invocation AcquireRelease|UniformMemory
+        };
+    }
+    }
+}
+__glsl_version(430)
+__spirv_version(1.0)
+[require(shadermemorycontrol)]
+[ForceInline]
+public void memoryBarrierBuffer()
+{
+    __target_switch
+    {
+    case hlsl: __intrinsic_asm "DeviceMemoryBarrier()";
+    case glsl: __intrinsic_asm "memoryBarrierBuffer()";
+    case spirv:
+    {
+        spirv_asm
+        {
+            OpCapability Shader;
+            OpMemoryBarrier Invocation AcquireRelease|UniformMemory
+        };
+    }
+    }
+}
+
+__glsl_version(430)
+__spirv_version(1.0)
+[require(any_gfx_target, shadermemorycontrol_compute)]
+[ForceInline]
+public void memoryBarrierShared()
+{
+    __target_switch
+    {
+    case hlsl: __intrinsic_asm "GroupMemoryBarrier()";
+    case glsl: __intrinsic_asm "memoryBarrierShared()";
+    case spirv:
+    {
+        spirv_asm
+        {
+            OpCapability Shader;
+            OpMemoryBarrier Workgroup AcquireRelease|WorkgroupMemory
+        };
+    }
+    }
+}
+__glsl_version(430)
+__spirv_version(1.0)
+[require(shadermemorycontrol)]
+[ForceInline]
+public void memoryBarrierImage()
+{
+    __target_switch
+    {
+    case hlsl: __intrinsic_asm "DeviceMemoryBarrier()";
+    case glsl: __intrinsic_asm "memoryBarrierImage()";
+    case spirv:
+    {
+        spirv_asm
+        {
+            OpMemoryBarrier Invocation AcquireRelease|ImageMemory
+        };
+    }
+    }
+}
+__glsl_version(430)
+__spirv_version(1.0)
+[require(any_gfx_target, shadermemorycontrol_compute)]
+[ForceInline]
+public void groupMemoryBarrier()
+{
+    __target_switch
+    {
+    case hlsl: __intrinsic_asm "GroupMemoryBarrier()";
+    case glsl: __intrinsic_asm "groupMemoryBarrier()";
+    case spirv:
+    {
+        spirv_asm
+        {
+            OpMemoryBarrier Workgroup AcquireRelease|WorkgroupMemory|ImageMemory|UniformMemory
+        };
+    }
+    }
+}
+
+/// Section 8.18. Subpass-Input Functions
+
+${{{{
+struct SubpassLoadEntriesType
+{
+    const char* typeName;
+    const char* prefix;
+};
+static const SubpassLoadEntriesType kSubpassLoadEntries[] =
+    {
+        { "vec4", "" },
+        { "uvec4", "u" },
+        { "ivec4", "i" },
+    };
+for (const auto& subpassLoadEntry : kSubpassLoadEntries)
+{
+}}}}
+    [require(glsl_spirv, subpass)]
+    public typealias $(subpassLoadEntry.prefix)subpassInput = __SubpassImpl<
+        $(subpassLoadEntry.typeName),
+        0 // isMS
+    >;
+
+    [require(glsl_spirv, subpass)]
+    [__NoSideEffect]
+    [ForceInline]
+    public $(subpassLoadEntry.typeName) subpassLoad($(subpassLoadEntry.prefix)subpassInput subpass)
```

Comment:
Can we define this as a single generic function, instead of duplicating it in the meta `for` loop?